### DISCUSSION TITLE
Is it possible to send metadata during peer discovery?

### DISCUSSION THREAD

**Discussion by radiohertz** [libp2p/rust-libp2p#4045] [Q&A]
Hi,
I'm using mDNS as the method of peer discovery and i want to attach a bit of metadata. Is this possible by any of the peer discovery protocols provided by the library or does this require me to write a custom protocol?

**Comment by mxinden** [comment]
Not today. I suggest writing your own `NetworkBehaviour` or embedding the information e.g. in the agent-version of `libp2p-identify`.

**Comment by dariusc93** [comment]
Depending on what you define as "metadata`, you could use `libp2p-request-response` to send data that you wish to provide to the peer (they would need the same implementation to handle receiving and responding to the data). Other option would be to send it via pubsub (eg gossipsub or floodsub) upon receiving the discovery event and determining that they are subscribed to the topic you wish to send it over.

**Reply by thomaseizinger** [reply]
I think OP was trying to send metadata during the initial discovery already.

**Comment by thomaseizinger** [comment]
You could easily develop your own UDP-based protocol _like_ mDNS and embed your information in there.

Mind sharing your usecase?
